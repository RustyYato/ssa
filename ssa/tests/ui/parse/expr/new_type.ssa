let Option = union[T] {
    Some: T,
    None: (),
};
let Pair = struct[T, U] {
    first: T,
    second: U,
};
let Tag = enum {
    Some,
    None,
};
