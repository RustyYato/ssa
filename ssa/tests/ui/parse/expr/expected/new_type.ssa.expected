[
  Item {
    id: ItemId(1)
    kind: ItemKind::Let(Let {
        binding: Ident {
          id: IdentId(1)
          name: "Option"
        }
        ty: 
        value: Option::Some(Expr {
            id: ExprId(1)
            kind: ExprKind::Union(ExprUnion {
                params: [
                  TypeParam {
                    name: Ident {
                      id: IdentId(2)
                      name: "T"
                    }
                    bounds: (
                    )
                  }
                ]
                variants: [
                  Field {
                    name: Ident {
                      id: IdentId(3)
                      name: "Some"
                    }
                    ty: Type {
                      id: TypeId(1)
                      kind: TypeKind::Concrete(TypeConcrete {
                          name: Path {
                            segments: [
                              Ident {
                                id: IdentId(4)
                                name: "T"
                              }
                            ]
                          }
                          generics: [
                          ]
                        }
                      )
                    }
                  }
                  Field {
                    name: Ident {
                      id: IdentId(5)
                      name: "None"
                    }
                    ty: Type {
                      id: TypeId(2)
                      kind: TypeKind::Primitive(TypePrimitive::Unit)
                    }
                  }
                ]
              }
            )
          }
        )
      }
    )
  }
  Item {
    id: ItemId(2)
    kind: ItemKind::Let(Let {
        binding: Ident {
          id: IdentId(6)
          name: "Pair"
        }
        ty: 
        value: Option::Some(Expr {
            id: ExprId(2)
            kind: ExprKind::Struct(ExprStruct {
                params: [
                  TypeParam {
                    name: Ident {
                      id: IdentId(7)
                      name: "T"
                    }
                    bounds: (
                    )
                  }
                  TypeParam {
                    name: Ident {
                      id: IdentId(8)
                      name: "U"
                    }
                    bounds: (
                    )
                  }
                ]
                fields: [
                  Field {
                    name: Ident {
                      id: IdentId(9)
                      name: "first"
                    }
                    ty: Type {
                      id: TypeId(3)
                      kind: TypeKind::Concrete(TypeConcrete {
                          name: Path {
                            segments: [
                              Ident {
                                id: IdentId(10)
                                name: "T"
                              }
                            ]
                          }
                          generics: [
                          ]
                        }
                      )
                    }
                  }
                  Field {
                    name: Ident {
                      id: IdentId(11)
                      name: "second"
                    }
                    ty: Type {
                      id: TypeId(4)
                      kind: TypeKind::Concrete(TypeConcrete {
                          name: Path {
                            segments: [
                              Ident {
                                id: IdentId(12)
                                name: "U"
                              }
                            ]
                          }
                          generics: [
                          ]
                        }
                      )
                    }
                  }
                ]
              }
            )
          }
        )
      }
    )
  }
  Item {
    id: ItemId(3)
    kind: ItemKind::Let(Let {
        binding: Ident {
          id: IdentId(13)
          name: "Tag"
        }
        ty: 
        value: Option::Some(Expr {
            id: ExprId(3)
            kind: ExprKind::Enum(ExprEnum {
                variants: [
                  Ident {
                    id: IdentId(14)
                    name: "Some"
                  }
                  Ident {
                    id: IdentId(15)
                    name: "None"
                  }
                ]
              }
            )
          }
        )
      }
    )
  }
]
